features/apps/src/test/java/org/codice/ddf/features/apps/test/ITAppsFeatureFile.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.features.apps.test;
+
+import static org.codice.ddf.features.test.config.DistributionOptions.includeDependencyPropertiesFile;
+import static org.codice.ddf.features.test.config.DistributionOptions.kernelDistributionOption;
+import static org.codice.ddf.features.test.config.FeatureOptions.addBootFeatureOption;
+import static org.codice.ddf.features.test.config.FeatureOptions.addFeaturesToFeatureRepo;
+import static org.codice.ddf.features.test.config.PortOptions.defaultPortsOptions;
+import static org.codice.ddf.features.test.config.VmOptions.defaultVmOptions;
+import static org.ops4j.pax.exam.CoreOptions.options;
+
+import java.io.File;
+
+import javax.inject.Inject;
+
+import org.codice.ddf.features.test.FeatureInstallException;
+import org.codice.ddf.features.test.FeatureServiceWrapper;
+import org.codice.ddf.features.test.FeatureUninstallException;
+import org.codice.ddf.features.test.features.AppsFeatureFile;
+import org.codice.ddf.features.test.features.TestUtilitiesFeatureFile;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.ops4j.pax.exam.Configuration;
+import org.ops4j.pax.exam.Option;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+public class ITAppsFeatureFile {
```

Comment:
I like how these are placeholders for integration tests but the testing of the feature install and uninstall seems like something someone could easily skip when adding a new feature. Can we make this automatically test new features?

Reply:
I definitely agree. I'm going to take a look at parameterized tests and see if we can use that to automate the test creation so we can avoid both the boilerplate and the possibility  of forgetting to test a feature.